src/Data/LLVM/BitCode/Parse.hs

```diff
@@ -736,9 +772,14 @@ resolveStrtabSymbol (Strtab bs) start len =
 
 -- Finalize Monad --------------------------------------------------------------
 
+data FinalizeEnv = FinalizeEnv { parsedEnv :: Env
+                               , parsedMdTable :: ValueTable
+                               , parsedValueTable :: ValueTable
+                               }
```

Comment:
Please add some Haddocks to this `FinalizeEnv` data type.

Also, it strikes me that we now have multiple `*Env` data types: we have `FinalizeEnv` here, and we also have a plain `Env` data type. This feels rather confusing. I wonder if `Env` should be renamed to `ParseEnv` to better distinguish the two?

Reply:
Haddocks added.

The `Env` could be renamed, but the two names are distinct, and `FinalizeEnv` is really only referenced and used "internally" in that small portion of the file that small portion of the file that implements the `Finalize` monad, so I was choosing in favor of not adding size to this PR by a renaming.